mydiff.diff

```diff
@@ -0,0 +1,249 @@
++
++// |enabled (DEFAULT)
++// a|Doc is auto purged
++
++// App notified of “accessRemoved” if _Documentlistener_ registered
++// a|Doc remains in local database
++
++// App notified of “Rejected Rev(403)” if _Documentlistener_ registered
++
++// |===
++
++
++[#lbl-isgr]
++== Inter-Sync Gateway
++
++=== Access Revoked
++
++NOTE: This behavior is the *reverse* of that between {sgw-t} and Couchbase Lite -- see: <<lbl-sgw-cbl>>.
++
++// Document auto-purge behavior in inter-Sync Gateway replications following channel access revocation.
++
++By default, documents are *not* auto purged on the active sync gateway even if the user on the passive sync gateway loses channel access.
++
++You can opt-in to auto-purge behavior using the replicator level option `purge_on_removal` in the REST API -- see {rest-api-admin-isgr--xref-purge-on-removal}.
++
++Documents will then *be* auto-purged -- on active Sync Gateway nodes that have opted-in -- if they do not belong to *any* of the replicating user’s footnote:[pass:q,a[The _replicating user_ is the user on the _passive_ sync gateway cluster; the user specified in the replication definition.]] channels -- see: <<lbl-isgr-revoke-behaviors>>.
++
++
++// [#tbl-isgr-revoke-behaviors,cols="^1,^2,3,4"]
++// |===
++
++// .2+^.^h|Direction
++// 2+^h|System State
++// ^h|Impact on Sync
++
++// h|Active Sync Gateway (Local -- Running as admin user)
++// h|Passive Sync Gateway (Remote)
++// ^h|Expected behavior when `purge_on_removal=true`
++// .Access Regained behavior
++[#lbl-isgr-revoke-behaviors]
++.Access Revocation behavior
++====
++Access control policies are only enforced at the remote cluster.
++
++Here the Active Sync Gateway (Local) is running as an admin user with  `purge_on_removal=true`
++
++[#tbl-isgr-revoke-behaviors,cols="^1a,3a,6a", options="header"]
++|===
++
++// .2+^.^h
++|Direction
++|Passive Sync Gateway (Remote)
++|Expected Sync behavior
++
++|Pull
++|User revoked access to channel
++|Previously synced documents are auto purged on local
++
++|Push
++|User revoked access to channel
++|Previously synced documents are auto purged on local
++
++
++|PushAndPull
++|User revoked access to channel +
++Sync Function includes `requireAccess(“channel”)`
++|When access is revoked on remote, the previously synced documents for User2 are auto-purged on local.
++
++Local changes continue to  be pushed to remote but rejected by remote
++
++|===
++====
++
++
++=== Access Regained
++
++If a user subsequently regains access to a lost channel then any previously auto-purged documents still assigned to any of their channels are automatically pulled down by the active Sync Gateway  -- see: <<lbl-isgr-regain-behaviors>>.
++
++If you want to control whether to sync previous auto purged versions of documents (rather than pull down purged documents) then you must also move the documents out of all of the users' channels so they are not synced down again.
```

Comment:
Same as on CBL side, this is confusing? Not sure what this is supposed to mean.

Reply:
> Generally fine but not sure about the use of `requireAccess` in both the SGW and CBL docs.

It was included because it is repeatedly mentioned in PRD. I  have removed mention of the sync function in this context now.